sources/osg/Matrix.js

```diff
@@ -1223,6 +1223,34 @@ define( [
             };
         } )(),
 
+        scale: function ( x, y, z, m, res ) {
+            if ( res === undefined ) {
+                Notify.warn( 'no matrix destination !' );
+                res = Matrix.create();
+            }
+            // fill CPU cache L1
+            Matrix.copy( m, res );
```

Comment:
I think it's best if we stick with the glmatrix version...
Plus... we are writing twice in the res array (and we are reading twice the same values too, once in the copy call and once again with the ``res[ x ] *=`` ).